### ISSUE TITLE
Incorrect matching / folding when using the `Better Folding` extension

### ISSUE THREAD

**Issue by edgimar** [zokugun/vscode-explicit-folding#118]
## Describe the issue
When using an extension like `Better Folding`, it can happen that you end up with folds in your file that do not originate from the explicit folding extension.  When I define folds using the explicit folding extension, these are typically larger-scale folds that describe a kind of nested outline of the code, and I would like to be able to fold these defined folds independent of the folds coming from other sources.

I am using `explicitfolding.rules` with several nested `separatorRegex` rules to match lines staring with `//|` and having varying numbers of `*` characters to indicate nesting level.  Specifically, I am using this:
```
    "explicitFolding.rules": {
        "rust": {
            "separatorRegex": "\\s*\\/\\/\\| \\*(?=\\s|$)",
            "strict": "never",
            "nested": [
                {
                    "separatorRegex": "\\s*\\/\\/\\| \\*\\*(?=\\s|$)",
                    "nested": [
                        {
                            "separatorRegex": "\\s*\\/\\/\\| \\*\\*\\*(?=\\s|$)",
                            "nested": [
                                {
                                    "separatorRegex": "\\s*\\/\\/\\| \\*\\*\\*\\*(?=\\s|$)",
                                    "nested": [
                                        {
                                            "separatorRegex": "\\s*\\/\\/\\| \\*\\*\\*\\*\\*(?=\\s|$)",
                                        }
                                    ]
                                }
                            ]
                        }
                    ]
                }
            ]
        }
    },
```

Two concerns:

1. as mentioned above, it would be nice if there were commands available to fold/unfold (e.g. recursively) ONLY the folds that correspond to any of the regexes, ignoring all other sources of folding.
2. It appears that there is some kind of interaction with `Better Folding` that prevents the regex from reliably being detected. See the following image that demonstrates this:

![image](https://github.com/user-attachments/assets/bb656f8d-bd46-4ca0-be08-c34aee03b586)

**Comment by daiyam** [comment]
Have you tried to check if your config does what you want by using the setting `"editor.defaultFoldingRangeProvider": "zokugun.explicit-folding",` and disabling those extensions?

**Comment by edgimar** [comment]
> Have you tried to check if your config does what you want by using the setting `"editor.defaultFoldingRangeProvider": "zokugun.explicit-folding",` and disabling those extensions?

Yes, that is what I'm currently doing, but I see it as a workaround, since it shouldn't be required to disable the folding provided by rust-analyzer in order to use explicit folding (which appears to be what the effect of setting `editor.defaultFoldingRangeProvider` is).

**Comment by daiyam** [comment]
Can you give me an example?

**Comment by daiyam** [comment]
+1
